dlt/common/typing.py

```diff
@@ -438,6 +438,10 @@ def get_generic_type_argument_from_instance(
     """
     orig_param_type = Any
     if cls_ := getattr(instance, "__orig_class__", None):
+        # unfurl Optional[Incremental[...]] to Incremental[...]
+        if is_optional_type(cls_):
```

Comment:
you could also look at `extract_inner_type` in `common/typing.py`. it will unfurl other wrappers (ie. Annotated) and also nested cases

Reply:
Nice! Got rid of the conditional check and just used `extract_inner_type` instead.